apps/examples/mediarecorder/mediarecorder_main.cpp

```diff
@@ -234,29 +245,36 @@ class MediaRecorderTest : public MediaRecorderObserverInterface
 			return;
 		}
 
-		mp.setObserver(shared_from_this());
-		mp.setDataSource(std::move(source));
-
-		if (mp.prepare() == PLAYER_OK) {
-			isPlaying = true;
-			if (mp.start() == PLAYER_OK) {
-				// playback started
-				return;
-			}
+		if (mp.setObserver(shared_from_this()) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setObserver failed" << std::endl;
+			goto errout_destroy;
+		}
 
-			std::cout << "Mediaplayer::start failed" << std::endl;
-			isPlaying = false;
+		if (mp.setDataSource(std::move(source)) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setDataSource failed" << std::endl;
+			goto errout_destroy;
+		}	
 
-			if (mp.unprepare() == PLAYER_ERROR) {
-				std::cout << "Mediaplayer::unprepare failed" << std::endl;
-			}
-		} else {
+		if (mp.prepare() == PLAYER_ERROR) {
 			std::cout << "Mediaplayer::prepare failed" << std::endl;
+			goto errout_destroy;
 		}
 
+		if (mp.start() == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::start failed" << std::endl;
+			goto errout_unprepare;
+		}
+		return;
+
+	errout_unprepare:
```

Comment:
Instead of goto, why don't you call unprepare directly?
Also if you want to call destroy, then we'd better add command for destroy because it supports create already in APP_ON

Reply:
It applied the comment of Chan-hee
I agree with you.
I will reverse it.